mlflow/utils/model_utils.py

```diff
@@ -191,3 +194,78 @@ def _validate_onnx_session_options(onnx_session_options):
                 raise ValueError(
                     f"Value for key {key} in onnx_session_options should be >= 0, not {value}"
                 )
+
+
+def _get_overridden_pyfunc_model_config(
+    pyfunc_config: Dict[str, Any], load_config: Dict[str, Any], logger
+) -> Dict[str, Any]:
+    """
+    Updates the inference configuration according to the model's configuration and the overrides.
+    Only arguments already present in the inference configuration can be updated. The environment
+    variable ``MLFLOW_PYFUNC_INFERENCE_CONFIG`` can also be used to provide additional inference
+    configuration.
+    """
+
+    overrides = {}
+    if env_overrides := os.getenv("MLFLOW_PYFUNC_INFERENCE_CONFIG"):
+        logger.debug(
+            "Inference configuration is being loaded from ``MLFLOW_PYFUNC_INFERENCE_CONFIG``"
+            " environ."
+        )
+        overrides.update(dict(json.loads(env_overrides)))
+
+    if load_config:
+        overrides.update(load_config)
+
+    if not overrides:
+        return pyfunc_config
+
+    if not pyfunc_config:
+        logger.warning(
+            f"Argument(s) {', '.join(overrides.keys())} were ignored since the model's ``pyfunc``"
+            " flavor doesn't accept model configuration. Use ``model_config`` when logging"
+            " the model to allow it."
+        )
+
+        return None
+
+    allowed_config = {key: value for key, value in overrides.items() if key in pyfunc_config.keys()}
+
+    if len(allowed_config) < len(overrides):
+        ignored_keys = set(overrides.keys()) - set(allowed_config.keys())
+        logger.warning(
+            f"Argument(s) {', '.join(ignored_keys)} were ignored since they are not valid keys in"
+            " the corresponding section of the ``pyfunc`` flavor. Use ``model_config`` when"
+            " logging the model to include the keys you plan to indicate. Current allowed"
+            " configuration includes"
+            f" {', '.join(pyfunc_config.keys())}"
+        )
+
+    pyfunc_config.update(allowed_config)
+    return pyfunc_config
```

Comment:
Could we do set operations instead of dict length comparisons to be more exact on the keys being overridden?

```suggestion
  valid_keys = set(pyfunc_config.keys()) & set(overrides.keys())
  ignored_keys = set(overrides.keys()) - valid_keys
  allowed_config = {key: overrides[key] for key in valid_keys}
  if ignored_keys:
      logger.warning(
          f"Argument(s) {', '.join(ignored_keys)} were ignored since they are not valid keys in"
          " the corresponding section of the ``pyfunc`` flavor. Use ``model_config`` when"
          " logging the model to include the keys you plan to indicate. Current allowed"
          f" configuration includes {', '.join(pyfunc_config.keys())}"
      )
  pyfunc_config.update(allowed_config)
  return pyfunc_config
```